src/services/rfqm_service.ts

```diff
@@ -0,0 +1,95 @@
+import { MarketOperation, QuoteRequestor } from '@0x/asset-swapper';
+import { BigNumber } from '@0x/utils';
+
+import { RFQT_REQUEST_MAX_RESPONSE_MS } from '../config';
+import { NULL_ADDRESS } from '../constants';
+
+export interface FetchIndicativeQuoteParams {
+    apiKey: string;
+    amount: BigNumber; // in terms of the amount the taker is selling
+    buyToken: string;
+    sellToken: string;
+    takerAddress?: string;
+}
+
+export interface FetchIndicativeQuoteResponse {
+    buyAmount: BigNumber;
+    buyTokenAddress: string;
+    price: BigNumber;
+    sellAmount: BigNumber;
+    sellTokenAddress: string;
+}
+
+/**
+ * RfqmService is the coordination layer for HTTP based RFQM flows
+ */
+export class RfqmService {
+    constructor(private readonly _quoteRequestor: QuoteRequestor) {}
+
+    /**
+     * Fetch the best indicative quote available.
+     */
+    public async fetchIndicativeQuoteAsync(params: FetchIndicativeQuoteParams): Promise<FetchIndicativeQuoteResponse> {
+        const { amount, sellToken, buyToken, apiKey } = params;
+
+        // Map params to the terminology of Quote Requestor
+        const marketOperation = MarketOperation.Sell;
```

Comment:
Are we not also supporting buys?

Reply:
Had supported both, but after addressing a previous comment, I removed it to only support sells: https://github.com/0xProject/0x-api/pull/535#discussion_r615048092

Sounds like this might be a discussion to have with the Taker team. I get the sense there is a push to frame things in terms of sells - but I lack context to chime in.